amundsen_application/static/js/components/TableDetail/RequestMetadataForm/index.tsx

```diff
@@ -0,0 +1,110 @@
+import * as React from 'react';
+import './styles.scss';
+
+import { sendNotification } from 'ducks/notification/api/v0';
+import { NotificationType } from 'interfaces/Notifications';
+import { AvatarLabelProps } from 'components/common/AvatarLabel';
+import { GlobalState } from 'ducks/rootReducer';
+import { connect } from 'react-redux';
+
+interface StateFromProps {
+  userEmail: string;
+  tableName: string;
+  tableOwners: Array<String>;
+}
+
+type RequestMetadataProps = StateFromProps;
+
+interface RequestMetadataState {
+  isOpen: boolean,
+}
+
+export class RequestMetadataForm extends React.Component<RequestMetadataProps, RequestMetadataState> {
+  public static defaultProps: RequestMetadataProps = {
+    userEmail: '',
+    tableName: '',
+    tableOwners: [],
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isOpen: true,
+    };
+  }
+
+  toggle = () => {
+    this.setState({ isOpen: !this.state.isOpen });
+  }
+
+  submitForm = (event) => {
+    event.preventDefault();
+    const form = document.getElementById("RequestForm") as HTMLFormElement;
+    const formData = new FormData(form);
+    const recipients = String(formData.get('recipients')).split(",");
+    const sender = String(formData.get('sender'));
+    const descriptionRequested = formData.get('table-description') === "on" ? true : false;
+    const fieldsRequested = formData.get('column-description') === "on" ? true : false;
+    const comment = String(formData.get('details'));
+    sendNotification(
+      recipients,
+      sender,
+      NotificationType.REQUESTED,
+      {
+        comment,
+        resource_name: this.props.tableName,
+        resource_url: window.location.href,
+        description_requested: descriptionRequested,
+        fields_requested: fieldsRequested,
+      }
+    )
+  };
+
+  render() {
+    const expandedClass = this.state.isOpen ? 'expanded' : 'collapsed';
+    return (
+      <div className={`request-component ${expandedClass}`}>
+        <div className="form-section request-header">
+          <h3 className="title">Amundsen Resource Request</h3>
+          <button type="button" className="btn btn-close" aria-label={"Close"} onClick={this.toggle}/>
+        </div>
+        <form onSubmit={ this.submitForm } id="RequestForm">
```

Comment:
You should separate this out into a separate component for testing when you get to that point, and even futher consider creating subcomponents for each form section. 

The reason is that unit testing the render methods of components with so much pure html makes updating unit tests very tedious. Why? Because when we test that each item is rendered we tend to write tests that equate to "Check if the nth child is this thing". With so much pure html combined into one render method, in the future if we decide we want to add some other component in, it will push the indices of all the components after it up by one, and then we have to update the indices for all of those tests as well.

I will let you decide if you want to do this now, or as a cleanup task when the majority of work is complete. If later, be sure to file a ticket so we don't forget.

Reply:
^ will do this as a followup task